crates/library/std/src/context.fe

```diff
@@ -5,6 +5,15 @@ use ingot::error::{
     Error
 }
 
+struct OutOfReachMarker {}
+
+// ctx.emit(my_event) should be the only way to emit an event. We achieve this by defining the
+// private `OutOfReachMarker` here to which only the `Context` has access.
+// Now there is no way to call `emit` directly on an Emittable.
+pub trait Emittable {
+  fn emit(self, _ val: OutOfReachMarker);
+}
```

Comment:
```suggestion
pub trait Emittable {
  fn emit(self, _ ctx: Context);
}
```
I think exposing private types from a public function is not a good idea especially when it's a trait function, how about receiving `Context` instead of the `OutOfReachMarker` type?

Reply:
>how about receiving `Context` instead of the `OutOfReachMarker` type

I thought about that as well but that would mean that people can bypass the `ctx.emit(..)` API and instead call `MyEvent().emit(ctx)`. That may be fine (since they are still depending on the `Context` but I kinda like it more if we force them to go through `ctx.emit(..)`. But I don't have very strong feelings on this.